hydrogym/core.py

```diff
@@ -355,19 +422,43 @@ def step(
     """Advance the state of the environment.  See gym.Env documentation
 
         Args:
-            action (Iterable[ArrayLike], optional): Control inputs. Defaults to None.
+            action (Iterable[ActType], optional): Control inputs. Defaults to None.
 
         Returns:
-            Tuple[ArrayLike, float, bool, dict]: obs, reward, done, info
+            Tuple[ObsType, float, bool, dict]: obs, reward, done, info
         """
-    self.solver.step(self.iter, control=action)
-    self.iter += 1
-    t = self.iter * self.solver.dt
+    action = action * self.flow.CONTROL_SCALING
+
+    if self.num_sim_substeps_per_actuation is not None and self.num_sim_substeps_per_actuation > 1:
+      self.action = action
+      self.flow = self.solver.solve_multistep(
+          num_substeps=self.num_sim_substeps_per_actuation,
+          callbacks=[self.rewardLogCallback],
+          controller=self.constant_action_controller,
+          start_iteration_value=self.iter)
+      if self.reward_aggreation_rule == "mean":
+        averaged_objective_values = np.mean(self.flow.reward_array, axis=0)
```

Comment:
This `reward_array` doesn't seem to be a consistent property or attribute of the `flow` class unless the `RewardCallback` is used.  Besides which, I would argue it shouldn't be an attribute of the flow at all.

Reply:
an attribute of the corresponding wrapper then